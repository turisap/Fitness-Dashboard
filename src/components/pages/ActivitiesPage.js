/**
 * Created by HP on 17-Jan-18.
 */
import React from 'react';
import {Link} from 'react-router-dom'

const ActivitiesPage = props => (
    <Link to="/"><h1>Home</h1></Link>
);

export default ActivitiesPage;