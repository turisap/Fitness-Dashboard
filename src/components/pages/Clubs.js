/**
 * Created by HP on 21-Jan-18.
 */
import React from 'react';
import { connect } from 'react-redux';

import {getAthletesClubs} from '../../actions/clubs';
import Club from '../Club';
import Loader from '../Loader';
import Fade from '../MountAnimation';


/**
 * Represents the clubs page
 */
class Clubs extends React.Component {

    state = {
        nothingWasFound : false
    };

    /**
     * Makes an AJAX request to STRAVA API if there is no clubs in the Redux store
     */
    componentDidMount () {
        if(!this.props.clubs.length){
            this.props.getClubs();
        }
        setTimeout(() => {
            if(!this.props.clubs.length) this.setState({nothingWasFound :true})
        },5000)
    }


    render() {
        return(
            <div>
                {!this.state.nothingWasFound && (this.props.clubs.length > 0
                    ?
                    this.props.clubs.map(cl => <Club key={cl.id} club={cl}/>)
                    :
                    <Loader/>
                )}
                <Fade in={this.state.nothingWasFound}>
                     <p>No clubs were found.. Try again later</p>
                </Fade>
            </div>
        )
    }
}



const mapDispatchToProps = dispatch => ({
    getClubs : () => dispatch(getAthletesClubs()),
});

const mapStateToProps = state => ({
    clubs : state.clubs.athleteClubs,
});

export default connect(mapStateToProps, mapDispatchToProps)(Clubs)