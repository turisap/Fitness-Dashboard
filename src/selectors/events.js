/**
 * Created by HP on 05-Jan-18.
 */

/**
 * Returns an exact event (via id) from the events or filtered events arrays in Redux store
 * @param state
 * @param id
 * @returns {Array.<T>}
 */
export const findEventById = (state, id) => {
    let event = state.events.filteredEvents.filter(ev => ev.id === id);
    if(!event.length) event = state.events.events.filter(ev => ev.id === id);
    return event;
};