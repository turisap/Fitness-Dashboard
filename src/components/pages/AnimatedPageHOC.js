/**
 * Created by HP on 23-Dec-17.
 */
/**
 * This is a HOC for animating pages on initial mount (all animations are specified in one place)
 * It also get user's location for further usage in other components
 */

import React from 'react';
import {connect} from 'react-redux';
import {CSSTransitionGroup} from 'react-transition-group';

import {setUserLocation} from '../../actions/userData';
import {setError} from '../../actions/events';

export default (AnimatedPage) => {
    class Animation extends React.Component {

        /**
         * Gets user's location for further usage in all page components
         */
        componentDidMount() {
            const getLocation = () => {
                const geolocation = navigator.geolocation;

                return  new Promise((resolve, reject) => {
                    if (!geolocation) {
                        reject(new Error('Not Supported'));
                    }

                    geolocation.getCurrentPosition((position) => {
                        //console.log(position.coords);
                        resolve(position);
                    }, () => {
                        reject(new Error('Permission denied'));
                    });
                });
            };

            getLocation()
                .then(resp => this.props.setLocation(resp.coords))
                .catch(setError('Location wasn\'t obtained'))

        }
        /*
        * Pipes each page through animation on initial mount
        * */
        render() {
            return (
                <CSSTransitionGroup
                    transitionName="mount-animation"
                    transitionAppear={true}
                    transitionAppearTimeout={500}
                    transitionEnter={false}
                    transitionLeave={false}>
                    <AnimatedPage {...this.props}/>
                </CSSTransitionGroup>
            )
        }
    }

    const mapDispatchToProps = dispatch => ({
        setLocation : (location) => dispatch(setUserLocation(location))
    });



    return connect(undefined, mapDispatchToProps)(Animation);
}
