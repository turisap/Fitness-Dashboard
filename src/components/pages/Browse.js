/**
 * Created by HP on 23-Dec-17.
 */
import React from 'react';
import {connect} from 'react-redux';
import PropTypes from 'prop-types'


import FiltersForm from '../FiltersForm';
import FiltersBar from '../FiltersBar';
import Event from '../Event';
import {getFilteredEvents} from '../../actions/events';
import {getUsersLocality} from '../../actions/userData';
import {filtersAreSame} from '../../helpers/compareFilters';
import LoadingOrElements from '../LoadingOrElements';




/**
 * Represents the Browse page
 */
export class Browse extends React.Component {

    /**
     * Starts an AJAX request to obtain filtered events in accordance with user's input
     * Gets events baised on user's location on initial loac
     */
    componentDidMount() {
        if(this.props.events.length === 0)this.props.getFilteredEvents(this.props);
    }



    /**
     * Gets filtered events accordingly with user's input
     * Dispatches a new AJAX request if filters were changed
     * @param nextProps
     */
    componentDidUpdate(nextProps){
        this.props.getUsersLocality(nextProps.userData.location);
        if (!filtersAreSame(nextProps.filters, this.props.filters)) {
            this.props.getFilteredEvents(this.props);
        }
    }

    /**
     * Returns a boolean if there are any filters present
     * @param categoryName
     * @param typeOfE
     * @param textFilter
     * @param dateRangeText
     * @param price
     * @returns {*}
     */
    static shouldDisplayFilterBar({categoryName,typeOfE,textFilter, dateRangeText, price}) {
        return categoryName || typeOfE || textFilter || dateRangeText || price;
    }



    render(){
        const events = this.props.events.map(ev => <Event key={ev.id}  event={ev}  />);

        return(
            <div className="container-fluid browse-page">
                <div className="browse-page__filter-form">
                    <FiltersForm/>
                </div>
                <div className="browse-page__events">
                    {Browse.shouldDisplayFilterBar(this.props.filters)
                    ?
                        <FiltersBar/>
                    :
                        <div className="filter-fix"></div>
                    }
                    <div className="browse--page__events-container">
                        {LoadingOrElements(events, this.props.events.length)}
                    </div>
                </div>
            </div>
        )
    }
}


/**
 * Validation of props
 * @type {{userData, filters}}
 */
Browse.propTypes = {
    userData : PropTypes.object.isRequired,
    filters  : PropTypes.object.isRequired,
};


/**
 * Mapping filters and user's data to props
 * @param state
 */
const mapStateToProps = state => ({
    filters      : state.filters,
    userData     : state.userData,
    events       : state.events.filteredEvents
});


/**
 * Mapping dispatching of filtered events AJAX and user locality AJAX to google API
 * @param dispatch
 */
const mapDispatchToProps = dispatch => ({
    getFilteredEvents : args => dispatch(getFilteredEvents(args)),
    getUsersLocality  : args => dispatch(getUsersLocality(args))
});

export default connect(mapStateToProps, mapDispatchToProps)(Browse);


