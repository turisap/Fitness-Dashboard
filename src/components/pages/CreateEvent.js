/**
 * Created by HP on 23-Dec-17.
 */
import React from 'react';
import moment from 'moment';
import { Input } from 'antd';
import {connect} from 'react-redux';
import {saveEvent} from '../../actions/events';

import DateTimeSet from '../DateTimeSet';
import FileUpload from '../FileUpload';
import Validator from '../../helpers/fieldValidator';


/**
 * Represents create event page
 */
export class CreateForm extends React.Component{
    state = {
        touched : false,
        title : {
            value : '',
            error : ''
        },
        location : {
            value : '',
            error : ''
        },
        startDate : {
            value : '',
            error : ''
        },
        endDate   : {
            value : '',
            error : ''
        },
        picture : {
            value : '',
            error : ''
        },
        description : {
            value : '',
            error : ''
        },
        organizerName : {
            value : '',
            error : ''
        }

    };

    /**
     * Custom validation of user's input
     * @param e
     */
    handleSubmit = e => {
        e.preventDefault();
        const newState = Validator.validate(this.state);
        this.setState({...newState, touched : true});
        if(!newState.hasErrors) this.props.saveEvent(newState);
    };


    /**
     * Checks whether flag on a new event saving was changed and redirect user to homepage if it was
     * @param nextProps
     */
    componentWillReceiveProps(nextProps) {
        if(nextProps.newEventSaved != this.props.newEventSaved) this.props.history.push('/');
    }


    render(){
        const { TextArea } = Input;
        return (
            <div className="create-event-container">
                <form className="create-event__form" onSubmit={this.handleSubmit}>
                    <div className="create-event__fieldset">
                        <label htmlFor="title">Event Title</label>
                        <Input id="title" name="title" type="text" onChange={(e) => this.setState({title:{value:e.target.value}})}/>
                        {this.state.touched && this.state.title.error && <p className="error">{this.state.title.error}</p>}
                    </div>
                    <div className="create-event__fieldset">
                        <label htmlFor="location">Location</label>
                        <Input name="location" id="location" type="text" onChange={e => {this.setState({location:{value:e.target.value}})}}/>
                        {this.state.touched && this.state.location.error && <p className="error">{this.state.location.error}</p>}
                    </div>
                    <div className="create-event__fieldset create-event__fieldset--pickers">
                        <div className="create-event__picker">
                            <label htmlFor="start-date">Start Date</label>
                            <DateTimeSet
                                onChange={e => {this.setState({startDate:{value:e.format('DD:MM:YYYY')}})}}
                                id="startDate"
                                name="start-date"
                                context="today"
                                defaultDate={moment()}
                                defaultTime={moment()}
                                type="text"
                            />
                            {this.state.touched && this.state.startDate.error && <p className="error">{this.state.startDate.error}</p>}
                        </div>
                        <div className="create-event__picker">
                            <label htmlFor="start-date">End Date</label>
                            <DateTimeSet
                                onChange={e => {this.setState({endDate:{value:e.format('DD:MM:YYYY')}})}}
                                name="end-date"
                                context="today"
                                defaultDate={moment()}
                                defaultTime={moment()}
                                type="text"
                            />
                            {this.state.touched && this.state.endDate.error && <p className="error">{this.state.endDate.error}</p>}
                        </div>
                    </div>
                    <div className="create-event__fieldset">
                        <FileUpload name="picture" id="picture"  onChange={e => {this.setState({picture : {value: e}})}}/>
                        {this.state.touched && this.state.picture.error && <p className="error">{this.state.picture.error}</p>}
                    </div>
                    <div className="create-event__fieldset">
                        <TextArea
                            id="description"
                            onChange={e => {this.setState({description:{value:e.target.value}})}}
                            name="description"/>
                        {this.state.touched && this.state.description.error && <p className="error">{this.state.description.error}</p>}
                    </div>
                    <div className="create-event__fieldset">
                        <label htmlFor="title">Organizer Name</label>
                        <Input
                            id="organizerName"
                            name="organizer_name"
                            type="text"
                            onChange={e => {this.setState({organizerName : {value:e.target.value}})}} />
                        {this.state.touched && this.state.organizerName.error &&
                        <p className="error">{this.state.organizerName.error}</p>}
                    </div>
                    {this.props.submissionErrors.length > 0 && this.props.submissionErrors.map((err, i) => {
                        return <div className="error" key={i}>{err}</div>
                    })}
                    <div className="create-event__fieldset">
                        <button type="submit" className="btn btn-default create-event__button">
                            Submit
                        </button>
                    </div>
                </form>
            </div>
        )
    }
}

const mapStateToProps = state => ({
    submissionErrors : state.events.errors,
    newEventSaved : state.events.newEventSaved
});

const mapDispatchToProps = dispatch => ({
    saveEvent : event => dispatch(saveEvent(event))
});


export default connect(mapStateToProps, mapDispatchToProps)(CreateForm);