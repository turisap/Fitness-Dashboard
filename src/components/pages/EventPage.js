/**
 * Created by HP on 04-Jan-18.
 */
import React from 'react';
import {connect} from 'react-redux';
import moment from 'moment';

import {getVenue, resetVenue} from '../../actions/events';
import {findEventById} from '../../selectors/events';
import Map from '../GoogleMap';
import ENV from '../../../ENV';


/**
 * Represent the Event page
 * @param props
 * @returns {XML}
 * @constructor
 */
export class EventPage extends React.Component {

    /**
     * Fulfils an AJAX request to get venue's info
     */
    componentDidMount() {
        this.props.setVenue(this.props.event[0].venue_id);
    }



    /**
     * Sets current venue to an empty object when user leaves the page
     */
    componentWillUnmount() {
        this.props.resetVenue();
    }



    render() {
        const { logo, name, is_free, description, start, end} = this.props.event[0];
        const googleMapsUrl = `https://maps.googleapis.com/maps/api/js?key=${ENV.googleAPI.keyAPI}&v=3.exp&libraries=geometry,drawing,places`;
        const {address_1, address_2, city, country, name : venue_name, latitude, longitude} = this.props.venue;

        return (
            <div className="event-page__background">
                <div className="event-page__info">
                    <div className="event-page__header">
                        <div className="event-page__picture-holder">
                            <img src={logo && logo.url} />
                        </div>
                        <div className="event-page__basic-info">
                            <div className="event-page__start">
                                {start && moment(start.local).format('MMM Do')}
                            </div>
                            <div className="event-page__item-name">{name && name.text}</div>
                            {is_free && <div className="event__price-types">FREE</div>}
                        </div>
                    </div>
                    <hr className="divider"/>
                    <div className="event-page__body">
                        <div className="event-page__description"><span>Description: </span>
                            {description.text ? description.text : ''}
                        </div>
                        <div className="event-page__venue">
                            <div className="event-page__venue-element">{address_1 && <p>{address_1}</p>}</div>
                            <div className="event-page__venue-element">{address_2 && <p>{address_2}</p>}</div>
                            <div className="event-page__venue-element">{city && <p>{city}</p>}</div>
                            <div className="event-page__venue-element">{country && <p>{country}</p>}</div>
                            <div className="event-page__venue-element">{venue_name && <p>{venue_name}</p>}</div>
                            <div className="event--page__date">
                                {start && moment(start.local).format('DD')} - {end && moment(end.local).format('DD MMM')}
                            </div>
                        </div>
                    </div>
                    {latitude &&
                    <div className="event-page__map">
                        <Map
                            longitude={longitude}
                            latitude={latitude}
                            isMarkerShown
                            googleMapURL={googleMapsUrl}
                            loadingElement={<div style={{ height: `100%` }} />}
                            containerElement={<div style={{ height: `400px` }} />}
                            mapElement={<div style={{ height: `100%` }} />}
                        />
                    </div>
                    }
                </div>
            </div>
        )
    }
}


const mapStateToProps = (state, props) => ({
    event : findEventById(state, props.match.params.id),
    venue : state.events.currentEventVenue
});



const mapDispatchToProps = dispatch => ({
    setVenue : id => dispatch(getVenue(id)),
    resetVenue : () => dispatch(resetVenue()),
});


export default connect(mapStateToProps, mapDispatchToProps)(EventPage);