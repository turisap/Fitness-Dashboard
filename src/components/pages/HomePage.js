/**
 * Created by HP on 23-Dec-17.
 */
import React from 'react';
import {connect} from 'react-redux';
import PropTypes from 'prop-types';

import {getEvents, setError} from '../../actions/events';
import {getUsersLocality} from '../../actions/userData';
import SearchForm from '../SearchForm';
import Event from '../Event';
import LoadingOrElements from '../LoadingOrElements';
import img1 from '../../assets/img/ticket.jpg';



export class HomePage extends React.Component {

    /**
     * Dispatches getting event action after receiving user's location
     */
    componentDidUpdate() {
        if (!this.props.events.length) {
            this.props.getEvents(this.props.location);
            this.props.getLocality(this.props.location);
        }
    }


    render(){
        const events = this.props.events.map(event => <Event key={event.id} event={event}/>);
        return(
            <div className="homepage container=fluid">
                <section className="homepage__slider">
                    <img src="scripts/assets/ticket.jpg"/>
                    <section className="homepage__searchform">
                        <SearchForm/>
                    </section>
                </section>
                <section className="homepage__greeting">
                    {this.props.location.latitude && <h1 className="display-2">All events in {this.props.locality}</h1>}
                </section>
                <div className="homepage__events">
                    {LoadingOrElements(events, this.props.events.length)}
                </div>
            </div>
        )
    }
}

/**
 * Validation of props
 * @type {{location: *, events: *}}
 */
HomePage.propTypes = {
    location : PropTypes.object,
    events   : PropTypes.array,
};


/**
 * Mapping user's location alogn with nearby events to props
 * @param state
 */
const mapStateToProps = state => ({
    location : state.userData.location,
    locality : state.userData.locality,
    events   : state.events.events
});


/**
 * Mapping dispatchers of getting nearby events and setting error to props
 * @param dispatch
 */
const mapDispatchToProps = dispatch => ({
    getEvents : (cords) => dispatch(getEvents(cords)),
    setError : (error) => dispatch(setError(error)),
    getLocality : (coord) => dispatch(getUsersLocality(coord))
});


export default connect(mapStateToProps, mapDispatchToProps)(HomePage);