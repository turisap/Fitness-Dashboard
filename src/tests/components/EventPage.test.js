/**
 * Created by HP on 08-Jan-18.
 */
import React from 'react';
import { shallow } from 'enzyme';
import {EventPage} from '../../components/pages/EventPage';
import {events} from '../fixtures/events';
import venue from '../fixtures/venue';

let wrapper, setVenue, resetVenue;

beforeEach(() => {
    setVenue = jest.fn();
    resetVenue = jest.fn();
    wrapper = shallow(<EventPage
        setVenue={setVenue}
        resetVenue={resetVenue}
        event={events}
        venue={venue}
    />, { lifecycleExperimental: true });
});


test('should render EventPage properly', () => {
    expect(wrapper).toMatchSnapshot();
});


test('should call setVenue() on component initial mount', () => {
    expect(setVenue).toHaveBeenCalledWith(events[0].venue_id);
});


test('should reset venue on component unmount', () => {
    wrapper.unmount();
    expect(resetVenue).toHaveBeenCalled();
});