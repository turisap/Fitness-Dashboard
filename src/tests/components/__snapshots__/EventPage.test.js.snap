// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EventPage properly 1`] = `
<div
  className="event-page__background"
>
  <div
    className="event-page__info"
  >
    <div
      className="event-page__header"
    >
      <div
        className="event-page__picture-holder"
      >
        <img
          src="https://img.evbuc.com/https%3A%2F%2Fcdn.evbuc.com%2Fimages%2F39195996%2F153837911724%2F1%2Foriginal.jpg?h=200&w=450&rect=0%2C77%2C960%2C480&s=90876175696ee174f46f504f2492f072"
        />
      </div>
      <div
        className="event-page__basic-info"
      >
        <div
          className="event-page__start"
        >
          Jan 6th
        </div>
        <div
          className="event-page__item-name"
        >
          Privacy Saturdays FREE WITH RSVP BEFORE 11:+ OPEN BAR Call 281.753.1498 to reserve a table
        </div>
        <div
          className="event__price-types"
        >
          FREE
        </div>
      </div>
    </div>
    <hr
      className="divider"
    />
    <div
      className="event-page__body"
    >
      <div
        className="event-page__description"
      >
        <span>
          Description: 
        </span>
      </div>
      <div
        className="event-page__venue"
      >
        <div
          className="event-page__venue-element"
        />
        <div
          className="event-page__venue-element"
        />
        <div
          className="event-page__venue-element"
        />
        <div
          className="event-page__venue-element"
        />
        <div
          className="event-page__venue-element"
        >
          <p>
            Utopia
          </p>
        </div>
        <div
          className="event--page__date"
        >
          06
           - 
          08 Apr
        </div>
      </div>
    </div>
    <div
      className="event-page__map"
    >
      <withScriptjs(withGoogleMap(Component))
        containerElement={
          <div
            style={
              Object {
                "height": "400px",
              }
            }
          />
        }
        googleMapURL="https://maps.googleapis.com/maps/api/js?key=AIzaSyDw5fwsSljNsyfchlULrBM4g984oGq8onY&v=3.exp&libraries=geometry,drawing,places"
        isMarkerShown={true}
        latitude="36.1798781"
        loadingElement={
          <div
            style={
              Object {
                "height": "100%",
              }
            }
          />
        }
        longitude="-115.1337015"
        mapElement={
          <div
            style={
              Object {
                "height": "100%",
              }
            }
          />
        }
      />
    </div>
  </div>
</div>
`;
